### PR TITLE
Added Largest rectangle

### LINKED ISSUE #161: Largest Rectange Area (Histogram)
I would like to code using stack for finding the largest area available. Here, I would input a set of heights of a building and a  number of buildings available.Then, find the largest area available or vacant.
I am a participant of GSSOC'21

### PR DESCRIPTION
**Pull request by sudipg4112001**
### Related issue:
Fixes #161 

Added the code in C++, the folder stack

### CODE REVIEW

**Review by krishna-NIT — CHANGES_REQUESTED** [review]
@sudipg4112001  **Really Appreciate your work within time,**
do few changes to make it more understandable
_Add Description at starting &
Add test case/output time and space complexity at Last
add necessary comments._

**Comment by sudipg4112001** [comment]
> @sudipg4112001 **Really Appreciate your work within time,**
> do few changes to make it more understandable
> _Add Description at starting & Add test case/output time and space complexity at Last add necessary comments._

All the changes should be made within the code or I need to make them in a separate file .. ??

**Comment by krishna-NIT** [comment]
> > @sudipg4112001 **Really Appreciate your work within time,**
> > do few changes to make it more understandable
> > _Add Description at starting & Add test case/output time and space complexity at Last add necessary comments._
> 
> All the changes should be made within the code or I need to make them in a separate file .. ??

do changes within code;
at starting & at ending
you can refer to a pull request which has been merged successfully in this upstream repo.

**Comment by sudipg4112001** [comment]
Okay.. I will make the changes very soon

**Comment by sudipg4112001** [comment]
Made the required changes. Please review.

**Review by krishna-NIT — CHANGES_REQUESTED** [review]
The description is Good, Keep it up @sudipg4112001 bro !! 👍🏻 
 but once check the readme of this repo, few changes also needed
update last 2 points at the end
image

**Comment by sudipg4112001** [comment]
Okay.. Give me some time, I will update those parts.

**Comment by sudipg4112001** [comment]
The updated the code. Please do give a review.

**Review by krishna-NIT — APPROVED** [review]
@plazzy99 @rudrakshi99 @nakul-19 Please Check it once, ok from my side & if Proper then Merge it @sudipg4112001 has given good efforts 👍🏻 , Good Going @sudipg4112001

**Review by plazzy99 — CHANGES_REQUESTED** [review]
Update readme for the same

**Comment by sudipg4112001** [comment]
> Update readme for the same

Same algorithm will have codes in different languages. So, how will I create a readme.. Could you please elaborate ?

**Comment by sudipg4112001** [comment]
Please, clarify my doubt. So that I can make the changes accordingly.

**Comment by sudipg4112001** [comment]
@plazzy99 @krishna-NIT  please clarify my doubt... It's been long you didn't respond.

**Comment by krishna-NIT** [comment]
> @plazzy99 @krishna-NIT please clarify my doubt... It's been long you didn't respond.

@sudipg4112001 , I think @plazzy99 wants to say that Update and Link Que in this Readme file with language
by Adding que at end
https://github.com/Algo-Phantoms/Algo-Tree/tree/main/Stack

He might be busy in another task, wait, He will Respond 👍🏻

**Comment by sudipg4112001** [comment]
@plazzy99 sir, if you have time please clarify my doubt. It's been long there is no response.

**Comment by rudrakshi99** [comment]
> > Update readme for the same
> 
> Same algorithm will have codes in different languages. So, how will I create a readme.. Could you please elaborate ?

Update the list in the Readme of stack folder.

**Comment by sudipg4112001** [comment]
> > > Update readme for the same
> > 
> > 
> > Same algorithm will have codes in different languages. So, how will I create a readme.. Could you please elaborate ?
> 
> Update the list in the Readme of stack folder.

Not clear.. please elaborate

**Comment by krishna-NIT** [comment]
> > > > Update readme for the same
> > > 
> > > 
> > > Same algorithm will have codes in different languages. So, how will I create a readme.. Could you please elaborate ?
> > 
> > 
> > Update the list in the Readme of stack folder.
> 
> Not clear.. please elaborate

@sudipg4112001 add this queue here https://github.com/Algo-Phantoms/Algo-Tree/blob/main/Stack/readme.md at bottom end

**Comment by sudipg4112001** [comment]
Made the required changes. It's showing conflicting files because of the link provided. Please do review.

**Review by krishna-NIT — APPROVED** [review]

**Review by krishna-NIT — CHANGES_REQUESTED** [review]
Put Largest_Rectangle.cpp  to Code/C++/Largest_Rectangle.cpp 
not to Stack/Largest_Rectangle.cpp

**Comment by sudipg4112001** [comment]
> Put Largest_Rectangle.cpp to Code/C++/Largest_Rectangle.cpp
> not to Stack/Largest_Rectangle.cpp

That is what I have made. Please do recheck

**Comment by krishna-NIT** [comment]
> > Put Largest_Rectangle.cpp to Code/C++/Largest_Rectangle.cpp
> > not to Stack/Largest_Rectangle.cpp
> 
> That is what I have made. Please do recheck

Capture

**Comment by krishna-NIT** [comment]
push files to correct folder Code/C++/

**Comment by sudipg4112001** [comment]
Please review, changes are made.

**Comment by krishna-NIT** [comment]
resolve conflicts

**Comment by sudipg4112001** [comment]
Done @krishna-NIT

**Review by krishna-NIT — APPROVED** [review]

File: Stack/readme.md [hunk 1]
```diff
@@ -28,3 +28,9 @@ element in a full stack is called **overflow**.
 -   Balanced Bracket Problem ----> [C++](/Code/C++/balanced_bracket.cpp)
 -   Stock Span Problem ----> [C++](/Code/C++/Stock_Span_Problem.cpp)
 
+ LR
```
Comment (tarun26091999) [inline Stack/readme.md:31]: remove this line

File: Stack/readme.md [hunk 2]
```diff
@@ -28,3 +28,9 @@ element in a full stack is called **overflow**.
...
+- Balanced Bracket Problem ----> [C++](/Code/C++/balanced_bracket.cpp)
+
+- Largest Rectangle ----> [C++](/Code/C++/Largest_Rectangle.cpp)
+
+ main
```
Comment (tarun26091999) [inline Stack/readme.md:36]: remove this also

File: Stack/readme.md [hunk 3]
```diff
@@ -28,3 +28,9 @@ element in a full stack is called **overflow**.
 -   Balanced Bracket Problem ----> [C++](/Code/C++/balanced_bracket.cpp)
 -   Stock Span Problem ----> [C++](/Code/C++/Stock_Span_Problem.cpp)
 
+ LR
+- Balanced Bracket Problem ----> [C++](/Code/C++/balanced_bracket.cpp)
+
+- Largest Rectangle ----> [C++](/Code/C++/Largest_Rectangle.cpp)
```
Comment (tarun26091999) [inline Stack/readme.md:34]: make uniform indentation.

**Review by tarun26091999 — CHANGES_REQUESTED** [review]
put the testcases one below another, same with the complexities and also use comments to explain your steps.

File: Stack/readme.md (hunk 2 above)
Comment (sudipg4112001) [reply Stack/readme.md:36]: > remove this also

both the lines ?

**Comment by sudipg4112001** [comment]
@tarun26091999 @krishna-NIT @rudrakshi99 @plazzy99  I have made the required changes suggested. Please review.

**Comment by sudipg4112001** [comment]
Please review this PR. 4 days passed by since I have made the changes.

**Review by tarun26091999 — APPROVED** [review]

**Pull request merged**